mobile/lib/widgets/settings/networking_settings/local_network_preference.dart

```diff
@@ -0,0 +1,257 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+import 'package:immich_mobile/providers/auth.provider.dart';
+import 'package:immich_mobile/providers/network.provider.dart';
+
+class LocalNetworkPreference extends HookConsumerWidget {
+  const LocalNetworkPreference({
+    super.key,
+    required this.enabled,
+  });
+
+  final bool enabled;
+
+  Future<String?> _showEditDialog(
+    BuildContext context,
+    String title,
+    String hintText,
+    String initialValue,
+  ) {
+    final controller = TextEditingController(text: initialValue);
+
+    return showDialog<String>(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: Text(title),
+        content: TextField(
+          controller: controller,
+          autofocus: true,
+          decoration: InputDecoration(
+            border: OutlineInputBorder(),
+            hintText: hintText,
+          ),
+        ),
+        actions: [
+          TextButton(
+            onPressed: () => Navigator.pop(context),
+            child: const Text('CANCEL', style: TextStyle(color: Colors.red)),
+          ),
+          TextButton(
+            onPressed: () => Navigator.pop(context, controller.text),
+            child: const Text('SAVE'),
+          ),
+        ],
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final wifiNameText = useState("");
+    final localEndpointText = useState("");
+
+    useEffect(
+      () {
+        final wifiName = ref.read(authProvider.notifier).getSavedWifiName();
+        final localEndpoint =
+            ref.read(authProvider.notifier).getSavedLocalEndpoint();
+
+        if (wifiName != null) {
+          wifiNameText.value = wifiName;
+        }
+
+        if (localEndpoint != null) {
+          localEndpointText.value = localEndpoint;
+        }
+
+        return null;
+      },
+      [],
+    );
+
+    saveWifiName(String wifiName) {
+      wifiNameText.value = wifiName;
+      return ref.read(authProvider.notifier).saveWifiName(wifiName);
+    }
+
+    saveLocalEndpoint(String url) {
+      localEndpointText.value = url;
+      return ref.read(authProvider.notifier).saveLocalEndpoint(url);
+    }
+
+    handleEditWifiName() async {
+      final wifiName = await _showEditDialog(
+        context,
+        "WiFi Name",
+        "Your Wi-Fi name",
+        wifiNameText.value,
+      );
+
+      if (wifiName != null) {
+        await saveWifiName(wifiName);
+      }
+    }
+
+    handleEditServerEndpoint() async {
+      final localEndpoint = await _showEditDialog(
+        context,
+        "Server Endpoint",
+        "http://local-ip:2283/api",
```

Comment:
```suggestion
        "http://local-ip:2283/api",
```
I think we should add it to the constants file. Also, I noticed that some text hasn't been added for translation.

Reply:
This is just a placeholder so there is no need I think